__tests__/app.test.js

```diff
@@ -64,6 +64,36 @@ describe("/api/articles",()=>{
                     })
                 })
     })
+    test("GET 200 and all articles of a topic (mitch) upon request", () => {
+        return request(app)
+            .get("/api/articles?topic=mitch")
+            .expect(200)
+                .then(({body})=>{
+                    const {articles} = body
+                    expect(articles.length).toBe(12)
+                    articles.forEach((article)=>{
+                        expect(article.topic).toBe('mitch')
+                    })
+                })
+    })
+    test("GET 200 and no articles when passed a valid but non-existent topic", () => {
+        return request(app)
+            .get("/api/articles?topic=toby")
+            .expect(200)
+                .then(({body})=>{
+                    const {articles} = body
+                    expect(articles.length).toBe(0)
+                })
+    })
+    test("GET 200 and all articles when passed anything other than a topic (ignores invalid queries)", () => {
```

Comment:
I think this should be a 400 since there's a problem with the request itself here, but again giving back all the articles is what you will want to do in the frontend, and you can implement that easily even if you get a 404/400 error in the backend